packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -221,10 +221,16 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface<
         if (state is SignInSuccess) {
           hubEvent = AuthHubEvent.signedIn(state.user.authUser);
         }
-        if (state is FetchAuthSessionFailure &&
-            (state.exception is UnauthorizedException ||
-                state.exception is AuthNotAuthorizedException)) {
-          hubEvent = AuthHubEvent.sessionExpired();
+        if (state is FetchAuthSessionSuccess) {
+          final exception = state.session.userPoolTokensResult.exception;
+          // TODO(Jordan-Nelson): Update list of exceptions once FetchAuthSession
```

Comment:
Rght now `AuthNotAuthorizedException` is thrown for an expired session, but it should be `SessionExpiredException`. I have a task to update this.

`UnauthorizedException` was in the list previously, so I left it for now. I'll look into what scenarios what is thrown in to see if this needs to be included when I take care of the task mentioned above.